dvc/progress.py

```diff
@@ -161,3 +165,28 @@ def format_dict(self):
             d["ncols_desc"] = d["ncols_info"] = 1
             d["prefix"] = ""
         return d
+
+
+class FsspecCallback(fsspec.Callback):
+    def __init__(self, fs, path_info, progress_bar):
+        self.fs = fs
+        self.path_info = path_info
+        self.progress_bar = progress_bar
+        super().__init__()
+
+    def set_size(self, size):
+        """``set_size()`` is an API that might be called with ``None`` if the
+        information is not already present on the caller. In that case, we'll
+        retrieve the size from an ``info()`` call."""
+        if size is None:
+            size = self.fs.info(self.path_info)["size"]
+        self.progress_bar.total = size
+        self.progress_bar.refresh()
+
+    def relative_update(self, inc=1):
+        self.progress_bar.update(inc)
+        self.progress_bar.refresh()
```

Comment:
In `tqdm`, `update()` decides when to refresh the progress bar, so you don't need to force it with `refresh`, might be slower otherwise. Same on the `absolute_update`.